lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+      File.open(lock_file_path, "w") do |f|
```

Comment:
I don't think it can hurt, updated to delete at end safely only if the file exists.